### PR TITLE
Use interpolated strings in DataGridView

### PR DESCRIPTION
**Pull request by halgab**
Contributes to #8203

This changes span all the DataGridView-related files. A lot of the changes are about StringBuilder removal.

## Test methodology

static code analysis

## Test environment

8.0.100-preview.1.23115.2

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8699)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs [hunk 1]
```diff
@@ -12,11 +12,11 @@
 
 namespace System.Windows.Forms
 {
-    [Designer("System.Windows.Forms.Design.DataGridViewDesigner, " + AssemblyRef.SystemDesign)]
+    [Designer($"System.Windows.Forms.Design.DataGridViewDesigner, {AssemblyRef.SystemDesign}")]
```
Comment (halgab) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs:15]: I wasn't sure if I should be changing attributes to be using interpolated strings. What does the team want for this case?

Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs:15]: Yes, please change attribute arguments as well - we had done it elsewhere, but never systematically.

Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs:15]: Thanks, I updated other attributes in those files.

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Comment by Tanya-Solyanik** [comment]
No crash dump or screenshots available.

**Unhandled exception. System.InvalidOperationException: Invoke or BeginInvoke cannot be called on a control until the window handle has been created.**
   at System.Windows.Forms.Control.MarshaledInvoke(Control caller, Delegate method, Object[] args, Boolean synchronous) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:line 6899
   at System.Windows.Forms.Control.**BeginInvoke**(Delegate method, Object[] args) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Control.cs:line 4629
   at System.Windows.Forms.**WindowsFormsSynchronizationContext**.Post(SendOrPostCallback d, Object state) in /_/src/System.Windows.Forms/src/System/Windows/Forms/WindowsFormsSynchronizationContext.cs:line 93
   at Xunit.Sdk.Utilities.SyncContextAwaiter.OnCompleted(Action continuation)
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AwaitUnsafeOnCompleted[TAwaiter](TAwaiter& awaiter, IAsyncStateMachineBox box)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_1(Object state)
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   at System.Threading.Thread.StartCallback()
Process terminated. Assertion Failed
Marshaling control should have created its handle in its ctor.
   at System.NoAssertContext.NoAssertListener.Fail(String message, String detailMessage) in /_/src/System.Windows.Forms.Primitives/tests/TestUtilities/NoAssertContext.cs:line 104
   at System.Diagnostics.TraceInternal.Fail(String message, String detailMessage)
   at System.Diagnostics.TraceInternal.TraceProvider.Fail(String message, String detailMessage)
   at System.Diagnostics.Debug.Fail(String message, String detailMessage)
   at System.Diagnostics.Debug.Assert(Boolean condition, String message)
   at System.Windows.Forms.WindowsFormsSynchronizationContext..ctor() in /_/src/System.Windows.Forms/src/System/Windows/Forms/WindowsFormsSynchronizationContext.cs:line 33
   at System.Windows.Forms.WindowsFormsSynchronizationContext.InstallIfNeeded() in /_/src/System.Windows.Forms/src/System/Windows/Forms/WindowsFormsSynchronizationContext.cs:line 150
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(msoloop reason, ApplicationContext context) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Application.ThreadContext.cs:line 1094
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoop(msoloop reason, ApplicationContext context) in /_/src/System.Windows.Forms/src/System/Windows/Forms/Application.ThreadContext.cs:line 966
   at System.Windows.Forms.Application.DoEvents() in /_/src/System.Windows.Forms/src/System/Windows/Forms/Application.cs:line 794
   at Xunit.Sdk.WinFormsSynchronizationContextAdapter.PumpTill(SynchronizationContext synchronizationContext, Task task)
   at Xunit.Sdk.ThreadRental.<>c__DisplayClass11_0.<CreateAsync>b__0()
   at System.Threading.Thread.StartCallback()

**Comment by halgab** [comment]
I thought this were intermittent bug failures as the previous commit tripped on the x86 release build. I noticed at least one other pull request that failed on those tests (#6945). I have to say I have absolutely no idea how my changes would affect only this leg on not the others...

**Review by lonitra — APPROVED** [review]
Nice!

**Comment by lonitra** [comment]
Hm.. This call stack is very peculiar and one that I haven't seen before.. We should keep an eye out for this. 
@halgab I do not think that has to do with your changes and is most likely due to something that occurred in our test infrastructure. We are currently working on investigating the cause for intermittent failures like this (https://github.com/dotnet/winforms/issues/8607), but that should not block this PR from going in. I will rerun the tests for this one it.

**Comment by Tanya-Solyanik** [comment]
@lonitra  - this is an old failure - `NoAssertContext` is not working.

**Pull request merged**